src/Step.cc

```diff
@@ -121,25 +121,21 @@ bool Step::execute(Context& context, CommChannel* comm, StepIndex index)
     install_timeout_and_termination_request_hook(lua, now, get_timeout(), index, comm);
 
     copy_used_variables_from_context_to_lua(context, lua);
-
     auto protected_result = lua.safe_script(get_script(), sol::script_pass_on_error);
+    copy_used_variables_from_lua_to_context(lua, context);
```

Comment:
Yep, you are right. Because in the previous code when an error is detected we don't pass the `Context` variables back. And after throwing an exception there is possibly a need to also examine the variable content.